ci/docker/deploy.Dockerfile

```diff
@@ -38,14 +38,14 @@ RUN pushd ${SOURCE}/miniapp && \
 
 # Prune and bundle binaries
 RUN mkdir ${BUILD}-tmp && cd ${BUILD} && \
-    export TEST_BINARIES=`ctest --show-only=json-v1 | jq '.tests | map(.command[0]) | .[]' | tr -d \"` && \
+    export TEST_BINARIES=`PATH=${SOURCE}/ci:$PATH ctest --show-only=json-v1 | jq '.tests | map(.command | .[] | select(contains("check-threads") | not)) | .[]' | tr -d \"` && \
```

Comment:
This change is related to the change from `${test_target_name}` to `$<TARGET_FILE:${test_target_name}>` in `DLAF_AddTest.cmake.

CMake/CTest assumes that the first element in the `COMMAND` for a test is an executable name. If CTest can't find the location of the executable the json output will _simply not have a `command` entry_. Therefore we have to make sure `check-threads` is found in `PATH` when we populate `TEST_BINARIES`.

Additionally, when we inject `check-threads` as a wrapper the test itself is now the first _argument_ to the executable. This means that the json output will _not_ expand a test target name in that position. This is why I now changed all uses of `${test_target_name}` in the `COMMAND` to use `$<TARGET_FILE:${test_target_name}>` so that the path is expanded already by CMake.

Finally, because `check-threads` is now in the `command` array of the tests I filter it out for `TEST_BINARIES`.

Let me know if and how you'd like to see this commented in the CMake config and/or Dockerfile.

Reply:
Not sure. That line was already obscure before and not commented.
I easily figured out the reason of the changes without the comment (the same changes appear in the codecov dockerfile which displays before this file in the PR...)
Does anyone think a comment is needed?